### ISSUE TITLE
URI option required but not used when socket is provided

### ISSUE THREAD

**Issue by mgwidmann** [vic/apollo-phoenix-websocket#25]
In the following code, the `uri` option is required to be passed in, even though its not used since the socket parameter is provided. I've used this to supply my own socket connection parameters.

```javascript
createNetworkInterface({
    uri: '/this-does-not-matter',
    Socket: () => { return socket; }
  });
```

**Comment by vic** [comment]
Oh right, thanks for reporting this, are you interested in sending an small PR ?

**Comment by mgwidmann** [comment]
Yeah I can take a look... All you'd want is just to not have to pass it in then right? Should be just something like `{ Socket: () => { return socket; } }` passed in
